### PR TITLE
use compileComponents() is not required with webpack

### PR DESCRIPTION
**Pull request by wKoza**
Since we use webpack, we don't have to use compileComponents(). Webpack transforms html and css file to inline template.

### CODE REVIEW

**Comment by wKoza** [comment]
You are right, i think that it's not necessary

**Comment by hansl** [comment]
Hi @wKoza @deebloo ,

These should still be in there. You're right that when using the @ngtools/webpack plugin we don't need those (mostly because there's no fetch calls). However, there's no hard performance implication for having those and they ARE best practices for every test.

We aim at aligning very closely with the Angular official style guide which recommends calling those for every async test.

Please also note that even if we use webpack internally, we still try to hide webpack from the user. This uses an implementation detail that we could switch in the future (or the user decides to stop using the CLI) and all the tests would be rendered invalid.

TL;DR: this is not broken, this does not impact performance, it's best practice, so we don't need to change it.

**Comment by wKoza** [comment]
Hi @hansl ,

> We aim at aligning very closely with the Angular official style guide which recommends calling those for every async test.

[Angular official says](https://angular.io/docs/ts/latest/guide/testing.html#!#compile-components)  :
> WebPack developers need not call compileComponents because it inlines templates and css as part of the automated build process that precedes running the test.

At which Angular official style guide you refer ?

**Comment by wKoza** [comment]
@hansl 

Furthermore, [compileComponents () is asynchronous](https://github.com/angular/angular-cli/blob/master/packages/%40angular/cli/blueprints/ng2/files/__path__/app/app.component.spec.ts#L6). So, we must use async () method like this:

  BeforeEach` (async (() => {
     TestBed.configureTestingModule ({
       Declarations: [BannerComponent], // declare the test component
     })
     .compileComponents (); // compile template and css
   }));

not 

BeforeEach (() => {
     TestBed.configureTestingModule ({
       Declarations: [BannerComponent], // declare the test component
     })
    TestBed.compileComponents (); // compile template and css
   });

I will open a issue.

**Comment by jasedwards** [comment]
Hi, angular.io now specifically states that if you use angular-cli you don't need to use compileComponents.  Can removing the usage of this in the component schematic be revisited?

https://angular.io/guide/testing#calling-compilecomponents
